keps/1833-metrics-for-local-queue/README.md

```diff
@@ -92,47 +88,59 @@ type ControllerManager struct {
+| False    | Specified           | Specified            | The selectors are disregarded, metrics will not be exposed.                                                        |
 
 ### List of metrics for Local Queues:
 
 In the first iteration, following are the list of metrics that would contain information on Local Queue statuses:
 
-| Metrics Name                   | Prometheus Type | Description                                                 | 
-|--------------------------------|-----------------|-------------------------------------------------------------|
-| local_queue_pending_workloads  | Gauge           | The number of pending workloads                             |
-| local_queue_reserved_workloads | Counter         | Total number of workloads in the LocalQueue reserving quota |
-| local_queue_admitted_workloads | Counter         | Total number of admitted workloads                          |
-| local_queue_resource_usage     | Gauge           | Total quantity of used quota per resource for a Local Queue |
+| Metrics Name                             | Prometheus Type | Description                                                                                         | 
+|------------------------------------------|-----------------|-----------------------------------------------------------------------------------------------------|
+| local_queue_pending_workloads            | Gauge           | The number of pending workloads.                                                                    |
+| local_queue_reserved_workloads_total     | Counter         | Total number of workloads in the LocalQueue reserving quota.                                        | 
+| local_queue_admitted_workloads_total     | Counter         | Total number of admitted workloads.                                                                 |
+| local_queue_resource_usage               | Gauge           | Total quantity of used quota per resource for a Local Queue.                                        |
+| local_queue_evicted_workloads_total      | Counter         | The total number of evicted workloads in Local Queue.                                               |
+| local_queue_admission_wait_time_seconds  | Histogram       | The time from when a workload got the quota reservation until admission in local queue.             |
+| local_queue_reserved_wait_time_seconds   | Histogram       | The time between a workload was created or re-queued until it got quota reservation in local queue. |
+| local_queue_wait_time_seconds            | Histogram       | Time taken to accept the resources from cluster queue.                                              |
+| local_admission_checks_wait_time_seconds | Histogram       | The time from when a workload got the quota reservation until admission.                            |
```

Comment:
Could you please clarify the differences between `local_queue_admission_wait_time_seconds` and `local_admission_checks_wait_time_seconds`? Maybe `local_queue_admission_wait_time_seconds` could describe time from creating (or requeueing) a Workload until admission - similarly to [the clusterqueue metrics](https://kueue.sigs.k8s.io/docs/reference/metrics/#clusterqueue-status). In that case it would cover the `local_queue_wait_time_seconds` metric usage.

Reply:
@PBundyra As you mentioned and looking into the code, looks like we would just need these 3:

1. local_queue_reserved_wait_time_seconds: The time between when a workload was created or re-queued until it got a quota reservation in the local queue.
2. local_queue_admission_wait_time_seconds: The time from when a workload got the quota reservation until it was admitted in the local queue.
3. local_queue_wait_time_seconds: Time taken to accept the resources from the cluster queue, which is going to be local_queue_reserved_wait_time_seconds + local_queue_admission_wait_time_seconds.

This would cover workloads lifecycle of: getting created --> Got reservation of quota --> Admitted.

Reply:
Thank you @varshaprasad96! Sounds good to me, the only change I suggest is naming. Let's stick with `local_admission_checks_wait_time_seconds` instead of `local_queue_admission_wait_time_seconds` as the second metric you mentioned; and `local_queue_admission_wait_time_seconds` instead of `local_queue_wait_time_seconds` as the third one.

That way it would be consistent with the clusterqueue metrics I linked above, and express more accurately what happens inside Kueue. So the final state for these metrics I suggest is:

1. `local_queue_reserved_wait_time_seconds`: [...]
2. `local_admission_checks_wait_time_seconds`: [...]
3. `local_queue_admission_wait_time_seconds`: [...]

Reply:
Sounds good! Have modified the naming as suggested.